fem/estimators.hpp

```diff
@@ -203,6 +202,135 @@ class ZienkiewiczZhuEstimator : public AnisotropicErrorEstimator
 };
 
 
+/** @brief The NewZienkiewiczZhuEstimator class implements the Zienkiewicz-Zhu
+    error estimation procedure [1,2] using face-based patches [3].
+
+    [1] Zienkiewicz, O.C. and Zhu, J.Z., The superconvergent patch recovery
+    and a posteriori error estimates. Part 1: The recovery technique.
+    Int. J. Num. Meth. Engng. 33, 1331-1364 (1992).
+
+    [2] Zienkiewicz, O.C. and Zhu, J.Z., The superconvergent patch recovery
+    and a posteriori error estimates. Part 2: Error estimates and adaptivity.
+    Int. J. Num. Meth. Engng. 33, 1365-1382 (1992).
+
+    [3] Bartels, S. and Carstensen, C., Each averaging technique yields reliable
+    a posteriori error control in FEM on unstructured grids. Part II: Higher
+    order FEM. Math. Comp. 71(239), 971-994 (2002)
+
+    The required BilinearFormIntegrator must implement the methods
+    ComputeElementFlux() and ComputeFluxEnergy().
+
+   COMMENTS:
+   *  The present implementation ignores all single-element patches corresponding
+      to boundary faces. This is appropriate for Dirichlet boundaries, but
+      suboptimal for Neumann boundaries. Reference 3 shows that a constrained
+      least-squares problem, where the reconstructed flux is constrained by the
+      Neumann boundary data, is appropriate to handle this case.
+      NOTE THAT THIS CONSTRAINED LS PROBLEM IS NOT YET IMPLEMENTED, so it is
+      possible that the local error estimates for elements on a Neumann boundary
+      are suboptimal.
+   *  The global polynomial basis used for the flux reconstruction is, by default,
+      aligned with the physical Cartesian axis. For patches with 2D elements, this
+      has been improved on so that the basis is aligned with the physical patch
+      orientation. Reorientation of the flux reconstruction basis is helpful to
+      maintain symmetry in the refinement pattern and could be extended to 3D.
+   *  This estimator is ONLY implemented IN SERIAL.
+   *  Anisotropic refinement is NOT YET SUPPORTED.
+
+ */
+class NewZienkiewiczZhuEstimator : public AnisotropicErrorEstimator
```

Comment:
Since this is not an `AnisotropicErrorEstimator` (not yet anyway) maybe it should be a subclass of `ErrorEstimator`